airflow/providers/delta/sharing/hooks/delta_sharing.py

```diff
@@ -0,0 +1,286 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import copy
+import json
+import os.path
+import re
+import sys
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+import requests
+from requests import exceptions as requests_exceptions
+from tenacity import RetryError, Retrying, retry_if_exception, stop_after_attempt, wait_exponential
+
+from airflow import __version__
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+from airflow.models import Connection
+
+if sys.version_info >= (3, 8):
+    from functools import cached_property
+else:
+    from cached_property import cached_property
+
+
+USER_AGENT_HEADER = {'user-agent': f'airflow-{__version__}'}
```

Comment:
nit: could we set this with a delta-sharing prefix:
USER_AGENT_HEADER = {'user-agent': f'delta-sharing-airflow-{__version__}'}

Reply:
This will be changed to match recent changes in user agent format

Reply:
what changes specifically?

Reply:
there is an unified telemetry format across all tools: https://github.com/apache/airflow/blob/main/airflow/providers/databricks/hooks/databricks_base.py#L147

Reply:
Thanks! I'll take a note on our side to know that this header will indicate the traffic from airflow connector.